src/export/StructureDefinitionExporter.ts

```diff
@@ -488,82 +885,72 @@ export class StructureDefinitionExporter implements Fishable {
+   * @throws {InvalidLogicalParentError} when Logical does not have valid parent
    */
-  exportStructDef(fshDefinition: Profile | Extension): StructureDefinition {
+  exportStructDef(fshDefinition: Profile | Extension | Logical | Resource): StructureDefinition {
     if (
       this.pkg.profiles.some(sd => sd.name === fshDefinition.name) ||
-      this.pkg.extensions.some(sd => sd.name === fshDefinition.name)
+      this.pkg.extensions.some(sd => sd.name === fshDefinition.name) ||
+      this.pkg.logicals.some(sd => sd.name === fshDefinition.name) ||
+      this.pkg.resources.some(sd => sd.name === fshDefinition.name)
     ) {
       return;
     }
 
-    const parentName = fshDefinition.parent || 'Resource';
+    const structDef = this.getStructureDefinition(fshDefinition);
 
-    if (fshDefinition.name === parentName) {
-      const result = this.fishForMetadata(parentName, Type.Resource);
-      throw new ParentDeclaredAsProfileNameError(
-        fshDefinition.name,
-        fshDefinition.sourceInfo,
-        result?.url
-      );
-    }
-
-    if (fshDefinition.id === parentName) {
-      const result = this.fishForMetadata(parentName, Type.Resource);
-      throw new ParentDeclaredAsProfileIdError(
-        fshDefinition.name,
-        fshDefinition.id,
-        fshDefinition.sourceInfo,
-        result?.url
-      );
-    }
-
-    const json = this.fishForFHIR(parentName);
-
-    // If we still don't have a resolution, then it's not defined
-    if (!json) {
-      throw new ParentNotDefinedError(fshDefinition.name, parentName, fshDefinition.sourceInfo);
-    } else if (fshDefinition instanceof Extension && json.type !== 'Extension') {
-      throw new InvalidExtensionParentError(
-        fshDefinition.name,
-        parentName,
-        fshDefinition.sourceInfo
-      );
-    }
-
-    const structDef = StructureDefinition.fromJSON(json);
     if (structDef.inProgress) {
       logger.warn(
         `The definition of ${fshDefinition.name} may be incomplete because there is a circular ` +
-          `dependency with its parent ${parentName} causing the parent to be used before the ` +
+          `dependency with its parent ${fshDefinition.parent} causing the parent to be used before the ` +
           'parent has been fully processed.',
         fshDefinition.sourceInfo
       );
     }
 
     structDef.inProgress = true;
 
+    // At this point, 'structDef' contains the parent's metadata and elements; therefore,
+    // transform the current 'structDef' from the parent to the new StructureDefinition to
+    // be exported.
+    // Reset the parent's 'structDef.elements' as required.
+    this.resetParentElements(structDef, fshDefinition);
+    // Reset the original parent's metadata to that for the new StructureDefinition
     this.setMetadata(structDef, fshDefinition);
+    // Apply the AddElementRules to create the new elements before any other processing
+    this.applyAddElementRules(structDef, fshDefinition);
```

Comment:
I think having this up here before `applyInsertRules` on line 948 will cause us issues. Because what if an `addElementRule` is inside the `RuleSet` pointed to by an `insertRule`. But in trying to test this I am seeing that an `addElementRule` cannot be on a `RuleSet`. I know @joepaquette that grammar was set before you got here. I can't remember if that was intentional, or if we should add that. @mint-thompson any thoughts?

Reply:
Good question! My initial thought is could we end up with a chicken & egg problem? Could we define a rule set that includes an element that has not been added yet if we process insertRules before addElementRules?

Reply:
I feel like we should support addElement rules in rulesets.  I could see a use case where someone has a common set of elements they want in every one of their logical models.

Reply:
So, I'll move lines 928-929 down below the line containing `applyInsertRules(fshDefinition, this.tank);`